mongomock/aggregate.py

```diff
@@ -498,15 +498,86 @@ def _handle_lookup_stage(in_collection, database, options):
     foreign_field = options['foreignField']
     local_name = options['as']
     foreign_collection = database.get_collection(foreign_name)
-    for doc in in_collection:
+    out_doc = copy.deepcopy(in_collection)
```

Comment:
No need to deep copy, we are only modifying the first level, and a deep copy can be quite long. Please leave out from that PR, I'll take care of it for the lookup stage.